### PR TITLE
Bug/aac missing timeline

### PR DESCRIPTION
**Pull request by dedobbin**
Fixes https://github.com/pdeljanov/Symphonia/issues/196

### CODE REVIEW

**Comment by pdeljanov** [comment]
Could you please test this with the following command: `cat /path/to/your/aac/file.aac | symphonia-play -`.

I don't think the seek will work if the input is not a seekable file! You will need to use the seek buffer in these cases.

**Comment by dedobbin** [comment]
Alright. As discussed, it now guarantees being rewindable , getting rid of the nasty panic and making it possible to set a max bytes to sample to approximate the n_frames

**Comment by pdeljanov** [comment]
Getting very good results with this last set of changes!

**Comment by dedobbin** [comment]
Alright, fixed all CI problems

**Review by pdeljanov — CHANGES_REQUESTED** [review]
This seems to cause errors when testing with `symphonia-check`. I'm assuming we're dropping a few initial packets for some reason...

File: symphonia-codec-aac/src/adts.rs [hunk 1]
```diff
@@ -258,3 +266,72 @@ impl FormatReader for AdtsReader {
...
+            if new_pos >= total_len {
+                break;
+            }
+
+            let res = source.seek(SeekFrom::Start(new_pos));
+            if res.is_err() {
+                break;
+            }
+
+            for _ in 0..=500 {
```
[the remark this answers is not in the document]
Comment (dedobbin) [reply symphonia-codec-aac/src/adts.rs]: Yeah that seems on the extreme side. I was not sure what numbers to plug in, i lowered it now and it still gives acceptable output to me. Let me know if this works for you or how we should go about finding a sweet spot.

**Review by pdeljanov — APPROVED** [review]

**Pull request merged**